community/consistency-check/src/main/java/org/neo4j/consistency/checking/RelationshipRecordCheck.java

```diff
@@ -19,32 +19,78 @@
  */
 package org.neo4j.consistency.checking;
 
+import org.neo4j.consistency.checking.cache.CacheAccess;
+import org.neo4j.consistency.checking.cache.CacheAccess.Client;
+import org.neo4j.consistency.checking.full.MultiPassStore;
 import org.neo4j.consistency.report.ConsistencyReport;
-import org.neo4j.consistency.store.DiffRecordAccess;
+import org.neo4j.consistency.report.ConsistencyReport.RelationshipConsistencyReport;
+import org.neo4j.consistency.statistics.Counts;
+import org.neo4j.consistency.store.DirectRecordReference;
 import org.neo4j.consistency.store.RecordAccess;
+import org.neo4j.consistency.store.RecordReference;
+import org.neo4j.helpers.ArrayUtil;
 import org.neo4j.kernel.impl.store.record.Record;
 import org.neo4j.kernel.impl.store.record.RelationshipRecord;
 import org.neo4j.kernel.impl.store.record.RelationshipTypeTokenRecord;
 
-class RelationshipRecordCheck
-        extends PrimitiveRecordCheck<RelationshipRecord, ConsistencyReport.RelationshipConsistencyReport>
+import static org.neo4j.helpers.ArrayUtil.union;
+
+public class RelationshipRecordCheck extends
+        PrimitiveRecordCheck<RelationshipRecord,ConsistencyReport.RelationshipConsistencyReport>
 {
-    RelationshipRecordCheck()
+    public static final int CACHE_SLOT_SOURCE_OR_TARGET = 0;
+    public static final int CACHE_SLOT_PREV_OR_NEXT = 1;
+    public static final int CACHE_SLOT_RELATIONSHIP_ID = 2;
+    public static final int CACHE_SLOT_REFERENCE = 3;
+    public static final long SOURCE = 0;
+    public static final long TARGET = -1;
+    public static final long PREV = 0;
+    public static final long NEXT = -1;
+
+    public RelationshipRecordCheck()
     {
-        super( RelationshipTypeField.RELATIONSHIP_TYPE,
-               NodeField.SOURCE, RelationshipField.SOURCE_PREV, RelationshipField.SOURCE_NEXT,
-               NodeField.TARGET, RelationshipField.TARGET_PREV, RelationshipField.TARGET_NEXT );
+        this( RelationshipTypeField.RELATIONSHIP_TYPE, NodeField.SOURCE, RelationshipField.SOURCE_PREV,
+                RelationshipField.SOURCE_NEXT, NodeField.TARGET, RelationshipField.TARGET_PREV,
+                RelationshipField.TARGET_NEXT );
     }
 
-    private enum RelationshipTypeField implements
-            RecordField<RelationshipRecord, ConsistencyReport.RelationshipConsistencyReport>,
-            ComparativeRecordChecker<RelationshipRecord, RelationshipTypeTokenRecord, ConsistencyReport.RelationshipConsistencyReport>
+    @SafeVarargs
+    RelationshipRecordCheck(
+            RecordField<RelationshipRecord,ConsistencyReport.RelationshipConsistencyReport>... fields )
     {
-        RELATIONSHIP_TYPE;
+        super( fields );
+    }
+
+    public static RelationshipRecordCheck relationshipRecordCheckPass1()
```

Comment:
`relationshipRecordCheckPass1` and `relationshipRecordCheckPass2` should be updated with more descriptive names